Separate Excel Reporter from HTML Reporter artifacts.

squash-merge opened the pull request:
Following content are added in a new worksheet in the ExcelReport:
1. Links to the page source.
2. Links to the screen capture of the page.
3. Parameters to the test method.
4. Other TestNG output logs.
5. The stacktrace of all the failures.

Also changes to disable HTML Reporter.

krmahadevan commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/ExcelReport.java:
```diff
@@ -110,6 +117,20 @@ public void generateReport(List<XmlSuite> xmlSuites, List<ISuite> suites, String
+     * 
+     * @param fileName The file name without any path.
+     */
+    public void setExcelFileName(String fileName) {
```
Minor observation : We could have leveraged `Preconditions.checkArgument`

squash-merge replied:
Addressed: Used `Preconditions.checkArgument`.

krmahadevan commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/TestCaseResult.java:
```diff
@@ -91,18 +95,70 @@ public void setITestResultobj(ITestResult iTestResult) {
         // if failed, then add error details
         if (iStatus == ITestResult.FAILURE) {
-            lstError.add(iTestResult.getThrowable().toString());
+            String stacktrace = StringUtils.defaultString(StringUtils.substringBetween(
```
I would suggest that you please break this down into individual calls. It becomes difficult to read and also difficult to debug as well.

squash-merge replied:
Addressed: Split the statement into two. First may return null, which is handled in the second.

krmahadevan commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/TestOutputReport.java:
```diff
@@ -0,0 +1,125 @@
+                col = row.createCell(colNum);
+
+                col.setCellStyle(Styles.getSubHeading2StyleThinBorder());
+                if (outputLines.length > 1) {
```
Can be changed to

```
String cellValue="";
if (outputLines.length > 1) {
     cellValue=outputColumn;
}
col.setCellValue(cellValue);
```

squash-merge replied:
Addressed: Moved the condition to declaration of `String outputColumn`.

krmahadevan commented on client/src/test/java/com/paypal/selion/reports/reporter/excelreport/ExcelReportTest.java:
```diff
@@ -59,6 +84,34 @@ public void testExcelReporter() {
+            } catch (IOException e) {
+                fail("Test excel-report-test failed due to IOException.");
+            } finally {
+                if (fileInputStream != null) {
```
Please use `IOUtils.closeQuietyly(fileInputStream);`

squash-merge replied:
Addressed: Used `IOUtils.closeQuietyly(fileInputStream)` as suggested.

krmahadevan commented on client/src/test/java/com/paypal/selion/reports/reporter/excelreport/ExcelReportTest.java:
```diff
@@ -59,6 +84,34 @@ public void testExcelReporter() {
+                workbook = new HSSFWorkbook(fileInputStream);
+                assertEquals(ReportSheetNames.values().length, workbook.getNumberOfSheets(),
+                        "Verify number of worksheets in ExcelReport");
+            } catch (FileNotFoundException e) {
```
I dont think we resort to this practice of catching exceptions and explicitly failing the tests. By default TestNG fails a test if it throws an exception. So we could perhaps clear up this part.

squash-merge replied:
Addressed: Reduced error handling and removed explicit fail calls.

krmahadevan commented on client/src/test/java/com/paypal/selion/reports/reporter/excelreport/ExcelReportTest.java:
```diff
@@ -35,6 +47,19 @@
+    private Path p = Paths.get(SUREFIRE_DEFAULT_REPORTS_DIR, excelReportFileName);
+
+    @BeforeMethod(groups = { "excel-report-test" })
+    public void removeExcelFileBeforeTest() {
```
We could change this to 

```
        if (Files.isRegularFile(p)) {
            FileUtils.deleteQuietly(p.toFile());
        }

```

squash-merge replied:
Addressed: Conditionally used `FileUtils.deleteQuietly` as suggested.

krmahadevan commented on client/src/main/java/com/paypal/selion/configuration/ListenerInfo.java:
```diff
@@ -48,8 +66,18 @@ public boolean isEnabled() {
+     * Returns boolean value of the JVM argument when available, else returns the {@code isEnabledWhenJvmAbsent}.
+     */
+    static boolean getBooleanValFromVMArg(String vmArgValue, boolean isEnabledWhenJvmAbsent) {
+        boolean flag = isEnabledWhenJvmAbsent;
```
This variable is being assigned but I dont see it being used.

squash-merge replied:
@krmahadevan, I did not get this comment. I have renamed member of `class ListenerInfo` and the method parameters to make its readability better.

squash-merge committed:
Separate Excel Reporter from HTML Reporter artifacts.

Following content are added in a new worksheet in the ExcelReport:
1. Links to the page source.
2. Links to the screen capture of the page.
3. Parameters to the test method.
4. Other TestNG output logs.
5. The stacktrace of all the failures.

Also changes to disable HTML Reporter.

 Date:      Thu Jun 25 12:07:04 2015 -0700

mach6 commented on client/src/main/java/com/paypal/selion/configuration/ListenerInfo.java:
```diff
@@ -33,24 +38,62 @@
+     * 
+     * @param vmArgValue
+     *      The VM argument name.
+     * @param isListenerEnabledWhenJvmAbsent
```
How can the JVM be absent in a java process? I suspect you mean to call this more like `defaultStateWhenNotDefined`

squash-merge replied:
Addressed: Used conventional names and comments.

mach6 commented on client/src/main/java/com/paypal/selion/configuration/ListenerInfo.java:
```diff
@@ -33,24 +38,62 @@
+    }
+
+    /**
+     * Returns boolean value of the JVM argument when available, else returns the {@code isEnabledWhenJvmAbsent}.
```
the `{@code` portion of this javadoc is incorrect

squash-merge replied:
Addressed: Updated the @{code block.

mach6 commented on client/src/main/java/com/paypal/selion/configuration/ListenerInfo.java:
```diff
@@ -33,24 +38,62 @@
+
+    /**
+     * Saves information about listener class which is enabled/disabled per its VM argument. When VM argument is
+     * unavailable, it's enabled/disabled is decided by the caller via (@code isEnabledWhenJvmAbsent}.
```
the {@code portion of this javadoc is incorrect

squash-merge replied:
Addressed: Updated the **@{code** block.

mach6 commented on client/src/main/java/com/paypal/selion/configuration/ListenerInfo.java:
```diff
@@ -33,24 +38,62 @@
+     *            - The Class name.
+     * @param jvmArgToParse
+     *            - The JVM argument that should be read to decide if the listener is to be enabled/disabled.
+     * @param isListenerEnabledWhenJvmAbsent
```
How can the JVM be absent in a java process? I suspect you mean to call this more like `defaultStateWhenNotDefined`

squash-merge replied:
Addressed: Used conventional names and comments.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/ExcelReport.java:
```diff
@@ -402,32 +434,42 @@ private void generateTCBasedData(ArrayList<TestCaseResult> allTestsList) {
             str.add(tcResult.getGroup().toString());
             str.add(String.valueOf(tcResult.getDurationTaken()));
-            str.add(tcResult.getsslink());
+//            str.add(tcResult.getsslink());
```
remove the commented code

squash-merge replied:
addressed: removed comment.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/ExcelReport.java:
```diff
@@ -19,6 +19,9 @@
 import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.nio.file.Files;
```
the import `java.io.File` is no longer used in this class

squash-merge replied:
Addressed: Removed unused import.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/ExcelReport.java:
```diff
@@ -30,6 +33,7 @@
 import java.util.Map.Entry;
 import java.util.logging.Level;
 
+import org.apache.commons.lang3.StringUtils;
```
let's stick with `org.apache.commons.lang.StringUtils` unless there is something specific you need in `lang3` here

squash-merge replied:
Addressed: Changed lang3 imports to lang for client code in another commit.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/ReportMap.java:
```diff
@@ -87,9 +87,13 @@ public void addToGeneratedReport(List<BaseReport<V>> lstCustomReport) {
         for (String indReport : this.reportData.keySet()) {
             if (iTypeOfReport == 0) {
                 b = new SummaryReport(indReport);
-            } else {
```
the previous way of formatting the `else` conditions is preferred.  Use `} else {` on the same line.

squash-merge replied:
Addressed: Changed to `} else {`.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/ReportMap.java:
```diff
@@ -87,9 +87,13 @@ public void addToGeneratedReport(List<BaseReport<V>> lstCustomReport) {
         for (String indReport : this.reportData.keySet()) {
```
Please parameterize `BaseReport` on line 86

squash-merge replied:
Addressed: Added a suppress for raw types.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/TestOutputReport.java:
```diff
@@ -0,0 +1,121 @@
+import java.util.List;
+
+import org.apache.commons.collections.ListUtils;
+import org.apache.commons.lang.StringUtils;
```
this import is not used.

squash-merge replied:
Addressed: Removed unused import.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/BaseReport.java:
```diff
@@ -21,6 +21,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.apache.commons.lang.StringUtils;
```
this import is not used.

squash-merge replied:
Addressed: Removed the import.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/ExcelReport.java:
```diff
@@ -232,8 +255,16 @@ private void createReportMap() {
+        
+        // TestCase Output Details Report
+        Map<String, List<List<String>>> fifthTestOutputSubReportMap = new LinkedHashMap<String, List<List<String>>>();
+        fifthTestOutputSubReportMap.put("Tests Output", tcOutputData);
```
Lets' call this tab `Test Output` instead.

squash-merge replied:
Addressed: This is the section header. Changed this to 'Test Output'. Will also update ReportSheetNames#TESTOUTPUTDETAILSREPORT to use 'Test Output'.

mach6 commented on client/src/main/java/com/paypal/selion/reports/reporter/excelreport/TestCaseResult.java:
```diff
@@ -40,11 +45,11 @@
     private int iStatus;
 
     public static final String NA = "NA";
+    public static final String newLine = "\n";
```
Sonar will complain here.. Constants should have all upppercase names

squash-merge replied:
Addressed: Changed to uppercase name.

squash-merge committed:
Changed org.apache.commons.lang3 imports to lang where applicable.

squash-merge committed:
Capture screenshot when there was an AssertionError.

mach6 commented:
this PR has been updated and merged.
